### PR TITLE
Support pathfinder_getTransactionStatus

### LINKED ISSUE #103: [DOCS] Fix `getTransactionReceiptBy` description

### LINKED ISSUE #104: Update CodingKeys uses to support Swift 6
```swift
try verifyTransactionType(container: container, codingKeysType: Self.CodingKeys)
```
This causes the following warning, should address this now to avoid errors once Swift 6 is released.
``` 
Expected member name or constructor call after type name; this will be an error in Swift 6
```

### LINKED ISSUE #105: Support `pathfinder_getTransactionStatus`
Per request, we should temporarily add `pathfinder_getTransactionStatus` endpoint. 
It should be removed and replaced by `starknet_getTransactionStatus` once #95 is resolved.

### PR DESCRIPTION
**Pull request by DelevoXDG**
Add `Provider.pathfinderGetTransactionStatus`

Closes #105
Closes #104 
Closes #103

### CODE REVIEW

File: Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift [hunk 1]
```diff
@@ -153,6 +153,15 @@ public class StarknetProvider: StarknetProviderProtocol {
         return result.transactionReceipt
     }
 
+    public func getPathfinderTransactionStatus(hash: Felt) async throws -> StarknetGatewayTransactionStatus {
```
Comment (drknzz) [inline Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift:156]: Maybe name it the same as JsonRpcMethod, pathfinderGetTransactionStatus?

Comment (DelevoXDG) [reply Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift:156]: I'm not sure> tt sounds odd and usually when someone wants to use getters they start typing `obj.get...` but I don't mind `pathfinderGetTransactionStatus` either.

File: Sources/Starknet/Providers/StarknetProviderProtocol.swift [hunk 2]
```diff
@@ -109,7 +109,16 @@ public protocol StarknetProviderProtocol {
     /// - Parameters:
     ///  - transactions: list of transactions to simulate
     ///  - blockId: block used to run the simulation
-    ///  - simulationFlags: a set of simulation flags
+    ///  - simulationFlags: a set of simulation
+
+    /// Get the gateway status of a submitted transaction
+    /// This method is supported only by Pathfinder nodes. Use on your own risk.
```
[the remark this answers is not in the document]
Comment (DelevoXDG) [reply Sources/Starknet/Providers/StarknetProviderProtocol.swift:115]: Inspired by offline chat, but I agree, maybe "risk" is a bit of an overkill in this context 😅

File: Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift (hunk 1 above)
Comment (drknzz) [reply Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift:156]: Just a suggestion, it's fine the way it is though.

**Review by drknzz — APPROVED** [review]

**Pull request merged**